File: Odoo/addons/hackpdm/models/hp_test.py
from odoo import models, fields, api

class hp_test(models.Model):
    #base fields
    _name = 'hp.test'
    _description = 'test'
    _inherit = 'hp.common.model'
    _parent_name = 'manytoone'
    _parent_store = True

    #fields
    binary = fields.Binary()
    boolean = fields.Boolean()
    character = fields.Char()
    dates = fields.Date()
    datetimes = fields.Datetime()
    floats = fields.Float()
    html = fields.Html()
    image = fields.Image()
    integer = fields.Integer()
    json = fields.Json()
    selection = fields.Selection([
        ('option_a', 'Option A'),
        ('option_b', 'Option B'),
        ('option_c', 'Option C'),
    ], string="Selection")

    text = fields.Text()

    parent_path = fields.Char(
        index=True,
        unaccent=False,
    )

    manytoone = fields.Many2one(comodel_name="hp.test", string="Parent")
    onetomany = fields.One2many("hp.test", 'manytoone', string="Children")
    manytomany = fields.Many2many('hp.test', 'hp_test_rel', 'left_id', 'right_id', string="many2many")

    currency_id = fields.Many2one(
        'res.currency',
        string='Currency',
        required=True,
        default=lambda self: self.env.company.currency_id.id
    )
    monetary = fields.Monetary(currency_field='currency_id')

File: Src/HackPDM.Infrastructure/Odoo/Models/HpTest.cs
using System.Collections;

using HackPDM.Core;
using HackPDM.Domain.OdooModels.Models;
using HackPDM.Infrastructure.Odoo.Models;
using HackPDM.Shared.GlobalData;
using HackPDM.Shared.OdooAttributes;
// Resharper disable InconsistentNaming

namespace HackPDM.Infrastructure.Odoo.Models;

[OdooModel(OdooDefaultsConstants.HP_TEST_NAME, OdooDefaultsConstants.HP_TEST)]
public partial class HpTest : HpBaseModelTransport<HpTest>, IHpTestModel
{
	[OdooProp(OdooFieldType.Binary		, "binary"	 )] public byte[]? binary		{ get; set; }
	[OdooProp(OdooFieldType.Boolean		, "boolean"	 )]	public bool? boolean		{ get; set; }
	[OdooProp(OdooFieldType.Char		, "character")]	public string? character	{ get; set; }
	[OdooProp(OdooFieldType.Date		, "dates"	 )]	public DateTime? dates		{ get; set; }
	[OdooProp(OdooFieldType.DateTime	, "datetimes")]	public DateTime? datetimes	{ get; set; }
	[OdooProp(OdooFieldType.Float		, "floats"	 )]	public float? floats		{ get; set; }
	[OdooProp(OdooFieldType.Html		, "html"	 )]	public string? html			{ get; set; }
	[OdooProp(OdooFieldType.Image		, "image"	 )]	public byte[]? image		{ get; set; }
	[OdooProp(OdooFieldType.Integer		, "integer"	 )]	public int? integer			{ get; set; }
	[OdooProp(OdooFieldType.Json		, "json"	 )]	public Hashtable? json		{ get; set; }
	[OdooProp(OdooFieldType.Monetary	, "monetary" )]	public decimal? monetary	{ get; set; }
	[OdooProp(OdooFieldType.Selection	, "selection")]	public string? selection	{ get; set; }
	[OdooProp(OdooFieldType.Text		, "text"	 )]	public string? text			{ get; set; }
	[OdooProp(OdooFieldType.Many2One	, "manytoone" )] public Many2One? many2one	{ get; set; }
	[OdooProp(OdooFieldType.One2Many	, "onetomany" )] public One2Many? one2many	{ get; set; }
    [OdooProp(OdooFieldType.Many2Many	, "manytomany")] public Many2Many? many2many { get; set; }
	
	IMany2One? IHpTestModel.many2one { get => (IMany2One?)many2one; set => many2one = (Many2One?)value; }
	IOne2Many? IHpTestModel.one2many { get => (IOne2Many?)one2many; set => one2many = (One2Many?)value; }
	IMany2Many? IHpTestModel.many2many { get => (IMany2Many?)many2many; set => many2many = (Many2Many?)value; }
}
